### ISSUE TITLE
Enforcing PEP8 Conventions

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#27]
Flake8 is a library for examining PEP8 violations within a code base. For those that don't know, PEP8 is a body of code composition rules that ensure the code is readable/maintainable and so that old/unused/hazardous conventions (like imports that can invisibly rewrite base functions; e.g. F403: "`from library import *`") are gradually removed from Python code library.

After changes introduced to builds from #25, we now have a Python3.6 test-build with Tox-Travis that evaluates whether a PR (both within the `xclim` and `tests` folders) is consistent with PEP8. As it is presently set, any violation of PEP8 is a failed build and a failed build from Travis will prevent code coverage checks (ie: if the build is failing, why bother seeing if coverage has decreased). *This is a little too strict and probably not optimal at this point.*

For that reason, we need to do the following:
- Write code like we normally do.
- Within a terminal (or via a test configuration in PyCharm - I can help set it up) run `$ flake8 xclim` and `$ flake8 tests` to find PEP8 violations.
- Fix the PEP8 violations **and/or** add specific PEP8 conventions to ignore.

If we're all using PyCharm, I can create a configuration library that highlights or marks as errors the conventions that we decide to enforce (we should strive to be strict with conventions, but I get it).

**TL;DR: I would like to know if there are any PEP8 conventions that we don't want to follow.** We can opt to ease up on PEP8 violations for now and be more strict in the future. The xclim library is fully PEP8 compliant but the `tests` as they are written break the following conventions:
- F403: Wildcard import (`from library import *`)
- F405: Function may be undefined or defined from a `*` import
- F841: Unassigned variable

Once the code base is either 100% compliant or we allow some wiggle room, builds will start reporting code coverage and whether they compile on Python 2.7, 3.4, 3.5, and 3.6.

**Comment by huard** [comment]
I suggest we start strict and see if that is a hindrance.
